### PR TITLE
[subplugin] Create initial nnstreamer trainer subplugin @open sesame 11/11 10:50

### PR DESCRIPTION
**Pull request by songgot**
[subplugin] Create initial nnstreamer trainer subplugin
- Create initial version
- Modify meson.build and meson_options.txt

**Self evaluation:**
1. Build test:   [X]Passed [ ]Failed [ ]Skipped
2. Run test:     [ ]Passed [ ]Failed [X]Skipped

Signed-off-by: hyunil park <hyunil46.park@samsung.com>

### CODE REVIEW

File: nnstreamer/tensor_trainer/tensor_trainer_nntrainer.hh [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  /**
+   * @brief send input data to NNTrainerTrain
+   * @param input input tensor memory imported input and label
+   */
+  void sendTensorData(const GstTensorMemory *input);
+
+  std::unique_ptr<NNTrainer::InputTensorsInfo> train_data, valid_data;
+
+private:
+  unsigned int batch_size;
```
Comment (jihochu) [inline nnstreamer/tensor_trainer/tensor_trainer_nntrainer.hh:91]: they needs doxy comments.

File: nnstreamer/tensor_trainer/tensor_trainer_nntrainer.hh [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+};
+
+/**
+ * @brief Input tensors data and information
+ */
+class InputTensorsInfo {
+public:
+  InputTensorsInfo(unsigned int num_of_samples, unsigned int num_of_inputs,
+                   unsigned num_of_labels);
+  unsigned int count;
```
Comment (jihochu) [inline nnstreamer/tensor_trainer/tensor_trainer_nntrainer.hh:108]: empty line between method and property.

File: nnstreamer/tensor_trainer/tensor_trainer_nntrainer.hh [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  unsigned int num_of_input_list;
+
+  std::string model_config;
+  std::unique_ptr<ml::train::Model> model;
+};
+
+/**
+ * @brief Input tensors data and information
+ */
+class InputTensorsInfo {
```
Comment (jihochu) [inline nnstreamer/tensor_trainer/tensor_trainer_nntrainer.hh:104]: Isn't there any method? then how about struct ?

File: nnstreamer/tensor_trainer/tensor_trainer_nntrainer.hh [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+ * @author Hyunil Park <hyunil46.park@samsung.com>
+ * @bug    No known bugs except for NYI items
+ */
+
+#include <model.h>
+#include <nnstreamer_plugin_api.h>
+#include <nnstreamer_plugin_api_filter.h>
+#include <vector>
+
+namespace NNTrainer {
```
Comment (lhs8928) [inline nnstreamer/tensor_trainer/tensor_trainer_nntrainer.hh:20]: Do InputTensorsInfo and NNTrainerTrain are placed in NNTrainer namespace?

**Comment by jijoongmoon** [comment]
plz fix the ci

**Comment by songgot** [comment]
Removed

**Pull request closed without merging**